systemFlake.nix

```diff
@@ -60,15 +62,28 @@ let
     {
       inherit (selectedNixpkgs) system;
       modules = [
-        ({ pkgs, lib, ... }: {
-          networking.hostName = hostname;
-
-          nixpkgs = {
-            inherit (selectedNixpkgs) overlays config system;
-          };
-
-          system.configurationRevision = lib.mkIf (self ? rev) self.rev;
-          nix.package = lib.mkDefault pkgs.nixUnstable;
+        ({ pkgs, lib, options, ... }: {
+          # 'mkMerge` to separate out each part into its own module
+          _type = "merge";
+          contents = [
+            (optionalAttrs (options ? networking) { 
+              networking.hostName = hostname; 
+            })
+
+            (if options ? nixpkgs then {
+              nixpkgs = {
+                inherit (selectedNixpkgs) overlays config system;
+              };
+            } else { _module.args.pkgs = selectedNixpkgs; })
+
+            (optionalAttrs (options ? system.configurationRevision) {
+              system.configurationRevision = lib.mkIf (self ? rev) self.rev;
+            })
+
+            (optionalAttrs (options ? nix) { 
+              nix.package = lib.mkDefault pkgs.nixUnstable;
+            })
+          ];
```

Comment:
What is the reasoning behind all these optionalAttrs? Is this meant to support non-nixpkgs based systems something?

Also, for example `options ? nix` does not check if `nix.package` exists - is this intended?

Reply:
So you can't use `mkIf` while checking if an option exists, I've done a fair bit of experimentation with this, while trying to make devos features optional. You also can't do `// optionalAttrs`, that causes infinite recursion. This way each option is technically considered its own module, which can be evaluated properly.
I can switch to actually check if the entire option exists. My reasoning was that if the `nix` namespace exists, then its likely for the sub options to exist, but I'll change to check for `nix.package` specifically.

related: https://github.com/divnix/devos/pull/226 and I got the `mkMerge` strategy from https://discourse.nixos.org/t/how-to-check-is-an-option-available/2919/2

Reply:
Ohh right and reason, is that these options won't try to be set if they don't exist, so yeah systems like robotnix, home-manager, nix-darwin could benefit from this. While not hurting existing nixos users.